### PR TITLE
[GLUTEN-6813][CH] Support soundex function

### PR DESCRIPTION
**Pull request by taiyang-li**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#6813)

## How was this patch tested?

Original spark test  "soundex function" in GlutenStringFunctionsSuite

### CODE REVIEW

**Review by exmy — DISMISSED** [review]

File: gluten-ut/spark35/src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala [hunk 1]
```diff
@@ -811,6 +811,7 @@ class ClickHouseTestSettings extends BackendTestSettings {
     .exclude("ParseUrl")
     .exclude("SPARK-33468: ParseUrl in ANSI mode should fail if input string is not a valid url")
     .exclude("FORMAT") // refer https://github.com/apache/incubator-gluten/issues/6765
+    .exclude("soundex unit test")
```
Comment (baibaichen) [inline gluten-ut/spark35/src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala:814]: why exclude soundex unit test ？

**Comment by baibaichen** [comment]
@exmy  where is the ut to test this funtion?

**Review by zzcclp — APPROVED** [review]
LGTM

**Review by baibaichen — APPROVED** [review]
LGTM

**Comment by taiyang-li** [comment]
image

**Pull request merged**